public/js/UX.js

```diff
@@ -1,8 +1,78 @@
 var smartDropXX = 50;
 var smartDropYY = 80;
 
+// Object stores the position of context menu
+var ctxPos = {
+    x: 0,
+    y: 0,
+    visible: false,
+};
+var dataPasted; // Stores the copy/cut data
+
+// Function hides the context menu
+function hideContextMenu() {
+    var el = document.getElementById('contextMenu');
+    el.style = 'opacity:0;';
+    setTimeout(() => {
+        el.style = 'visibility:hidden;';
+        ctxPos.visible = false;
+    }, 200); // Hide after 2 sec
+}
+
+// Function displays context menu
+function showContextMenu() {
+    if (layoutMode) return false; // Hide context menu when it is in Layout Mode
+    $('#contextMenu').css({
+        visibility: 'visible',
+        opacity: 1,
+        top: `${ctxPos.y}px`,
+        left: `${ctxPos.x}px`,
+    });
+    ctxPos.visible = true;
+    return false;
+}
+
+// Function is called when context item is clicked
+// eslint-disable-next-line no-unused-vars
+function menuItemClicked(id) {
+    hideContextMenu();
+
+    if (id === 0) {
+        document.execCommand('copy');
+    } else if (id === 1) {
+        document.execCommand('cut');
+    } else if (id === 2) {
+        // document.execCommand('paste'); it is restricted to sove this problem we use dataPasted variable
+        if (dataPasted) paste(dataPasted);
```

Comment:
Can you rename this to data copied? Also are you sure there is no way around this?

Reply:
These are some alternatives pasting:
* I can use the [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) however it not compatible with all the browser. 
* I can save the copied data to local storage and then retrieve while pasting, this hack will help us to copy data from the different project.

Which method should i use??

Reply:
I had a similar conclusion to use localStorage as at least within the browser it will work. However, I would like to confirm that CTR+C still works the same way as before correct? And that doesn't have any browser restrictions right. 

@jbox144 I agree, I expected the context menu to be quite complicated, I was impressed.

Reply:
"document.execCommand('paste') is not supported by browsers, because of security concerns as it might enable the script to read sensitive data (like passwords) from the clipboard."

Copy command is supported by every browser.

Cut command is supported by every browser except IE.

after adding the local storage CTR+C will work the same way.

Reply:
How about ctr+v?

Reply:
it will also work the same way.